### ISSUE TITLE
Identify imported Harvester cluster and apply label accordingly

### ISSUE THREAD

**Issue by yasker** [rancher/rancher#34880]
**Is your feature request related to a problem? Please describe.**
Current only the Harvester clusters imported in the `Virtualization Management` page will be recognized as a Harvester cluster. The Harvester cluster imported in `Cluster Management` won't be recognized as a Harvester cluster on the `Virtualization Management` page. It will be treated as a normal K8s cluster, which will cause confusion for the user.

Currently, UI is [adding the label](https://github.com/rancher/dashboard/issues/4244#issuecomment-925987029) `cluster.harvesterhci.io` to any clusters imported from the `Virtualization Management` page.

**Describe the solution you'd like**
[Instead of driving by the UI, the backend should identify the Harvester cluster and apply the label accordingly.](https://github.com/rancher/dashboard/issues/4244#issuecomment-925968631)

**Describe alternatives you've considered**
The alternative would be to keep it as it is right now but it will be a pretty mixed-up experience for the user.
Also, we are adding a banner to remind users not to import Harvester from the Cluster Management page but that's just for the time being.

**Additional context**
REF: https://github.com/rancher/dashboard/issues/4244

**Comment by deniseschannon** [comment]
Based on discussions that we've had about when and how to treat a Harvester cluster, I don't believe this use case makes sense.

If i import a harvester cluster into Cluster Management, I would expect to manage it k8s cluster and it shouldn't automatically convert it to a Harvester cluster type in Rancher and show up in the Virtualization Management page.

We should discuss this with PM.

**Comment by gaktive** [comment]
Pushing out to 2.6.2 for now until next steps are determined.

**Comment by yasker** [comment]
@deniseschannon I don't agree that Harvester should be ever managed as a normal k8s cluster. It will be confusing to the user. And we also agreed on we shouldn't expose the k8s underneath Harvester unless for debugging purposes.

**Comment by richard-cox** [comment]
Once this has been resolved I'd like to review the warning banner added in https://github.com/rancher/dashboard/pull/4269, hopefully that can then be removed.

**Comment by guangbochen** [comment]
as per discussion with the @yasker and the Harvester UI team @WuJun2016 @n313893254 ,  cc @gaktive @richard-cox  @nwmac 
- use `provider=harvester` as the only identity to distinguish the regular k8s cluster and the Harvester cluster.
- the `cluster.harvesterhci.io` label should only be used on the Virtualization Management page when the `provider` type is empty, e.g before the user complete importing the cluster into the Rancher (this is a temp solution for now before we address the design gap on the UX)

**Comment by yasker** [comment]
More context after discussion with Guangbo:
* UI should be able to identify the Harvester cluster by `provider=harvester`.
* However, there was a concern that a cluster being imported doesn't have the `provider` set yet, so it will show up on the `Cluster Management` page. So the UI team introduced `cluster.harvesterhci.io` label to help pinning down the cluster in the `Virtualization Management` page. 
* Though that introduced another problem: in the current logic, the cluster will be identified as a `Harvester` cluster if either `cluster.harvesterhci.io` is set or `provider == harvester`. However, if a cluster is Harvester or not should be determined by the backend alone. Otherwise there is a corner case that any normal k8s cluster imported in the `Virtualization Management` will be recognized as Harvester cluster then totally break due to it cannot show either Harvester UI in `Virtualization Management` nor show `Cluster Explorer` since it doesn't exist in the `Cluster Management` page.
* So the conclusion is: we will only use  `cluster.harvesterhci.io` label to identify where the cluster should be before the cluster was completed imported. Once the cluster's `provider` becomes non-empty, the `provider` field should be used as the only way to identify should the cluster be shown on `Virtualization Management` or `Cluster Management` page. If the cluster is still being imported so `provider` is empty, then the `cluster.harvesterhci.io` will be used to identify which page to show the cluster instead.

**Comment by StrongMonkey** [comment]
Corresponding UI changes that need tio be done after https://github.com/rancher/rancher/pull/35007 is merged:
1. UI needs to add the label `provider.cattle.io=harvester` to any clusters imported from the Virtualization Management page.
2. UI needs to switch to use label `provider.cattle.io=harvester` to identify harvester clusters.
3. UI need to set the following rule in fleet avoid to deploy to harvester cluster when choose to `All cluster in the workspace`
```
spec:
  targets:
  - clusterSelector:
      matchExpressions:
      - key: "provider.cattle.io"
        operator: In
        values:
        - harvester
```

**Comment by gaktive** [comment]
FYI @nwmac in case @richard-cox needs to jumps on this. @yasker to confirm first.

**Comment by yasker** [comment]
Thank you @StrongMonkey ! Confirm those are necessary UI changes.

One additional note: UI might notice that the `provider.cattle.io` label will change after the cluster was imported. The backend logic will only set/override the `provider.cattle.io` to the provider (e.g. `harvester`) once the cluster has completed the importing process. The backend logic will not override the label with an empty value. So `provider.cattle.io=harvester` can be applied by the UI in the `Virtualization Management` page before the cluster importing is completed to temporarily classify the cluster as a Harvester cluster and show up in the `Virtualization Management` page, and the label will be automatically updated by the backend if needed once the importing process was completed.

**Comment by richard-cox** [comment]
@yasker @StrongMonkey The UI work for both this issue and https://github.com/rancher/rancher/issues/34954 has now been completed in https://github.com/rancher/dashboard/pull/4337. I've tested locally and clusters imported via the Virtualization Management section are identified correctly. Also fleet now only installs to non-harvester clusters when targeted at 'all' clusters (note - the operator needed to be `NotIn` rather than `In`).

Is there anything else remaining in this or 34954 blocking testing?

**Comment by yasker** [comment]
@richard-cox thanks for the fix. I think we can move those two issues to testing now.

**Comment by yasker** [comment]
To test:
1. Import a Harvester cluster in Cluster Management page
    * Before the importing is completed, the cluster should show up on the Cluster Management page
    * After the importing is completed, the cluster should show up on the Virtualization Management page instead.
2. Import a Harvester cluster in Virtualization Management page
    * The cluster should always show up on the Virtualization Management page.
3. Import a normal K8S cluster (non-harvester) in the Virtualization Management page
    * Before the importing is completed, the cluster should show up on the Virtualization Management page
    * After the importing is completed, the cluster should show up on the Cluster Management page instead.

**Comment by slickwarren** [comment]
tested on v2.6-head (62c101a) as a standard user, cluster owner:
1. Import a Harvester cluster in Cluster Management page
    * Before the importing is completed, the cluster should show up on the Cluster Management page -- pass
    * After the importing is completed, the cluster should show up on the Virtualization Management page instead. -- pass
2. Import a Harvester cluster in Virtualization Management page
    * The cluster should always show up on the Virtualization Management page. -- pass
3. Import a normal K8S cluster (non-harvester) in the Virtualization Management page
    * Before the importing is completed, the cluster should show up on the Virtualization Management page -- pass
    * After the importing is completed, the cluster should show up on the Cluster Management page instead. -- pass
4. provision normal k8s cluster using harvester node driver -- should always show up on virtualization management page -- pass
